code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -190,27 +236,108 @@ void setName(Map<String, Object> inspect, Map<String, Object> data, Instance ins
     }
 
     void setNetwork(Map<String, Object> inspect, Map<String, Object> data, Instance instance) {
-        Network network = null;
-        Network rancherNetwork = networkDao.getNetworkForObject(instance, KIND_HOSTONLY);
-        String ip = getDockerIp(inspect);
-        if (StringUtils.isNotEmpty(ip) && isIpInNetwork(ip, rancherNetwork)) {
-            DataAccessor.fields(instance).withKey(FIELD_REQUESTED_IP_ADDRESS).set(ip);
-            network = rancherNetwork;
+        String networkMode = checkNoneNetwork(inspect);
+        if (setNetworkModeIfNotNull(instance, networkMode))
+            return;
+
+        String inspectNetMode = getInspectNetworkMode(inspect);
+
+        networkMode = checkHostNetwork(inspectNetMode);
+        if (setNetworkModeIfNotNull(instance, networkMode))
+            return;
+
+        networkMode = checkContainerNetwork(inspectNetMode, instance, data);
+        if (setNetworkModeIfNotNull(instance, networkMode))
+            return;
+
+        networkMode = checkBridgeOrManagedNetwork(inspectNetMode, inspect, data, instance);
+        if (setNetworkModeIfNotNull(instance, networkMode))
+            return;
+
+        log.warn("Could not determine network mode for container [externalId: {}]. Using none networking.", instance.getExternalId());
+        setNetworkModeIfNotNull(instance, NETWORK_MODE_NONE);
+    }
+
+    private String getInspectNetworkMode(Map<String, Object> inspect) {
+        Object tempNM = CollectionUtils.getNestedValue(inspect, "HostConfig", "NetworkMode");
+        String inspectNetMode = tempNM == null ? "" : tempNM.toString();
+        return inspectNetMode;
+    }
+
+    private boolean setNetworkModeIfNotNull(Instance instance, String networkMode) {
+        if (networkMode != null) {
+            DataAccessor.fields(instance).withKey(FIELD_NETWORK_MODE).set(networkMode);
+            return true;
+        }
+        return false;
+    }
+
+    private String checkNoneNetwork(Map<String, Object> inspect) {
+        Object netDisabledObj = CollectionUtils.getNestedValue(inspect, "Config", "NetworkDisabled");
+        if (netDisabledObj != null && netDisabledObj instanceof Boolean && (Boolean)netDisabledObj) {
```

Comment:
Nit: we can use "Boolean.TRUE.equals(Object)" structure here, and then you don't need extra not null/instanceof checks.